internal/hcsoci/resources.go

```diff
@@ -93,12 +100,16 @@ func ReleaseResources(ctx context.Context, r *Resources, vm *uvm.UtilityVM, all
 			// vm they belong to.
 			if all {
 				if err := r.resources[i].Release(ctx); err != nil {
-					return err
+					log.G(ctx).WithError(err).Error("failed to release container resource")
+					errs = append(errs, err.Error())
 				}
 			}
 		}
 	}
 	r.resources = nil
+	if len(errs) != 0 {
+		return fmt.Errorf("failed to release all resources: %s", strings.Join(errs, "\n"))
```

Comment:
I don't think we need to do this join of the error strings here. We can just return an error saying one or more resources failed to clean up. Users can take a trace if they need more info.

Reply:
Ok I thought this would be helpful. I'll change